File: mobile_app/app_pedidos/Objects/src/com/jq/pedidos/designerscripts/LS_frmlogin.java
package com.jq.pedidos.designerscripts;
import anywheresoftware.b4a.objects.TextViewWrapper;
import anywheresoftware.b4a.objects.ImageViewWrapper;
import anywheresoftware.b4a.BA;


public class LS_frmlogin{

public static void LS_general(java.util.LinkedHashMap<String, anywheresoftware.b4a.keywords.LayoutBuilder.ViewWrapperAndAnchor> views, int width, int height, float scale) {
anywheresoftware.b4a.keywords.LayoutBuilder.setScaleRate(0.3);
anywheresoftware.b4a.keywords.LayoutBuilder.scaleAll(views);
views.get("panellogin").vw.setLeft((int)((0d / 100 * width)));
views.get("panellogin").vw.setWidth((int)((100d / 100 * width) - ((0d / 100 * width))));
views.get("panellogin").vw.setTop((int)((0d / 100 * height)));
views.get("panellogin").vw.setHeight((int)((100d / 100 * height) - ((0d / 100 * height))));
views.get("btnconfiguracion").vw.setLeft((int)(0d));
views.get("btnconfiguracion").vw.setTop((int)((1.5d / 100 * height)));
views.get("btnconfiguracion").vw.setHeight((int)((7d / 100 * height)));
views.get("btnconfiguracion").vw.setWidth((int)((15d / 100 * width)));
views.get("panelconfiguracion").vw.setTop((int)((views.get("btnconfiguracion").vw.getTop())));
views.get("panelconfiguracion").vw.setLeft((int)((views.get("btnconfiguracion").vw.getLeft() + views.get("btnconfiguracion").vw.getWidth())));
views.get("panelconfiguracion").vw.setWidth((int)((views.get("panellogin").vw.getWidth())-(views.get("panelconfiguracion").vw.getLeft())-(2d / 100 * width)));
views.get("panelconfiguracion").vw.setHeight((int)((views.get("btnconfiguracion").vw.getHeight())+(3d / 100 * height)));
views.get("txturl").vw.setLeft((int)(0d));
views.get("txturl").vw.setTop((int)(0d));
views.get("txturl").vw.setWidth((int)((views.get("panelconfiguracion").vw.getWidth())));
views.get("txturl").vw.setHeight((int)((views.get("panelconfiguracion").vw.getHeight())+(1.5d / 100 * height)));
views.get("viewlogo").vw.setTop((int)((10d / 100 * height)));
views.get("viewlogo").vw.setWidth((int)((45d / 100 * width)));
views.get("viewlogo").vw.setHeight((int)((views.get("viewlogo").vw.getWidth())-(3d / 100 * width)));
views.get("viewlogo").vw.setLeft((int)((50d / 100 * width) - (views.get("viewlogo").vw.getWidth() / 2)));
views.get("label3").vw.setLeft((int)(0d));
views.get("label3").vw.setWidth((int)((100d / 100 * width)));
views.get("label3").vw.setTop((int)((views.get("viewlogo").vw.getTop() + views.get("viewlogo").vw.getHeight())));
views.get("label3").vw.setHeight((int)((9d / 100 * height)));
views.get("panel1").vw.setTop((int)((views.get("label3").vw.getTop() + views.get("label3").vw.getHeight())+(6d / 100 * height)));
views.get("panel1").vw.setLeft((int)((7d / 100 * width)));
views.get("panel1").vw.setHeight((int)((7.5d / 100 * height)));
views.get("panel1").vw.setWidth((int)((100d / 100 * width)-(views.get("panel1").vw.getLeft())*2d));
views.get("txtusuario").vw.setLeft((int)(0d));
views.get("txtusuario").vw.setTop((int)(0d));
views.get("txtusuario").vw.setWidth((int)((views.get("panel1").vw.getWidth())));
views.get("txtusuario").vw.setHeight((int)((views.get("panel1").vw.getHeight())+(1.5d / 100 * height)));
views.get("panel2").vw.setTop((int)((views.get("panel1").vw.getTop() + views.get("panel1").vw.getHeight())+(3d / 100 * height)));
views.get("panel2").vw.setLeft((int)((views.get("panel1").vw.getLeft())));
views.get("panel2").vw.setHeight((int)((views.get("panel1").vw.getHeight())));
views.get("panel2").vw.setWidth((int)((views.get("panel1").vw.getWidth())));
views.get("txtcontrasena").vw.setLeft((int)(0d));
views.get("txtcontrasena").vw.setTop((int)(0d));
views.get("txtcontrasena").vw.setWidth((int)((views.get("panel2").vw.getWidth())));
views.get("txtcontrasena").vw.setHeight((int)((views.get("txtusuario").vw.getHeight())));
views.get("btnlogin").vw.setTop((int)((views.get("panel2").vw.getTop() + views.get("panel2").vw.getHeight())+(2d / 100 * height)));
views.get("btnlogin").vw.setHeight((int)((views.get("panel1").vw.getHeight())));
views.get("btnlogin").vw.setLeft((int)((views.get("panel1").vw.getLeft())));
views.get("btnlogin").vw.setWidth((int)((views.get("panel1").vw.getWidth())));
views.get("btnmenuprincipal").vw.setTop((int)((views.get("btnlogin").vw.getTop() + views.get("btnlogin").vw.getHeight())+(2d / 100 * height)));
views.get("btnmenuprincipal").vw.setLeft((int)((views.get("panel1").vw.getLeft())));
views.get("btnmenuprincipal").vw.setWidth((int)((views.get("panel1").vw.getWidth())));
views.get("btnmenuprincipal").vw.setHeight((int)((7d / 100 * height)));
views.get("lblregistrarme").vw.setLeft((int)(0d));
views.get("lblregistrarme").vw.setTop((int)((views.get("btnmenuprincipal").vw.getTop() + views.get("btnmenuprincipal").vw.getHeight())+(1d / 100 * height)));
views.get("lblregistrarme").vw.setWidth((int)((100d / 100 * width)));
views.get("lblregistrarme").vw.setHeight((int)((5d / 100 * height)));
views.get("label2").vw.setHeight((int)((5d / 100 * height)));
views.get("label2").vw.setWidth((int)((100d / 100 * width)));
views.get("label2").vw.setLeft((int)(0d));
views.get("label2").vw.setTop((int)((100d / 100 * height) - (views.get("label2").vw.getHeight())));

}
}

File: mobile_app/app_pedidos_ios/Objects/src/B4iProject/LS_frmlogin.m

#import "iCore.h"
@interface LS_frmlogin:NSObject
@end

@implementation LS_frmlogin

- (void)LS_general:(B4ILayoutData*)views :(int)width :(int)height{
[B4ILayoutBuilder setScaleRate:0.3];
[B4ILayoutBuilder scaleAll:views :width :height];
[views get:@"imagenfondo"].Top = (int)(0.0);
[views get:@"imagenfondo"].Height = (int)((100 / 100.0 * height) - (0.0));
[views get:@"imagenfondo"].Left = (int)(0.0);
[views get:@"imagenfondo"].Width = (int)((100 / 100.0 * width) - (0.0));
[views get:@"btnconfiguracion"].Left = (int)(0.0);
[views get:@"btnconfiguracion"].Top = (int)((1.5 / 100.0 * height));
[views get:@"btnconfiguracion"].Height = (int)((7 / 100.0 * height));
[views get:@"btnconfiguracion"].Width = (int)((15 / 100.0 * width));
[views get:@"txturl"].Top = (int)(([views get:@"btnconfiguracion"].Top));
[views get:@"txturl"].Left = (int)(([views get:@"btnconfiguracion"].Left + [views get:@"btnconfiguracion"].Width));
[views get:@"txturl"].Width = (int)((100 / 100.0 * width)-([views get:@"txturl"].Left)-(2 / 100.0 * width));
[views get:@"txturl"].Height = (int)(([views get:@"btnconfiguracion"].Height)+(3 / 100.0 * height));
[views get:@"viewlogo"].Top = (int)((10 / 100.0 * height));
[views get:@"viewlogo"].Width = (int)((45 / 100.0 * width));
[views get:@"viewlogo"].Height = (int)(([views get:@"viewlogo"].Width)-(3 / 100.0 * width));
[views get:@"viewlogo"].Left = (int)((50 / 100.0 * width) - ([views get:@"viewlogo"].Width / 2));
[views get:@"label3"].Left = (int)(0.0);
[views get:@"label3"].Width = (int)((100 / 100.0 * width));
[views get:@"label3"].Top = (int)(([views get:@"viewlogo"].Top + [views get:@"viewlogo"].Height));
[views get:@"label3"].Height = (int)((9 / 100.0 * height));
[views get:@"panel1"].Top = (int)(([views get:@"label3"].Top + [views get:@"label3"].Height)+(3 / 100.0 * height));
[views get:@"panel1"].Left = (int)((7 / 100.0 * width));
[views get:@"panel1"].Height = (int)((7.5 / 100.0 * height));
[views get:@"panel1"].Width = (int)((100 / 100.0 * width)-([views get:@"panel1"].Left)*2.0);
[views get:@"txtusuario"].Left = (int)(0.0);
[views get:@"txtusuario"].Top = (int)(0.0);
[views get:@"txtusuario"].Width = (int)(([views get:@"panel1"].Width));
[views get:@"txtusuario"].Height = (int)(([views get:@"panel1"].Height)+(1.5 / 100.0 * height));
[views get:@"panel2"].Top = (int)(([views get:@"panel1"].Top + [views get:@"panel1"].Height)+(3 / 100.0 * height));
[views get:@"panel2"].Left = (int)(([views get:@"panel1"].Left));
[views get:@"panel2"].Height = (int)(([views get:@"panel1"].Height));
[views get:@"panel2"].Width = (int)(([views get:@"panel1"].Width));
[views get:@"txtcontrasena"].Left = (int)(0.0);
[views get:@"txtcontrasena"].Top = (int)(0.0);
[views get:@"txtcontrasena"].Width = (int)(([views get:@"panel2"].Width));
[views get:@"txtcontrasena"].Height = (int)(([views get:@"txtusuario"].Height));
[views get:@"btnlogin"].Top = (int)(([views get:@"panel2"].Top + [views get:@"panel2"].Height)+(4 / 100.0 * height));
[views get:@"btnlogin"].Height = (int)(([views get:@"panel1"].Height));
[views get:@"btnlogin"].Left = (int)(([views get:@"panel1"].Left));
[views get:@"btnlogin"].Width = (int)(([views get:@"panel1"].Width));
[views get:@"btnmenuprincipal"].Top = (int)(([views get:@"btnlogin"].Top + [views get:@"btnlogin"].Height)+(2 / 100.0 * height));
[views get:@"btnmenuprincipal"].Left = (int)(([views get:@"panel1"].Left));
[views get:@"btnmenuprincipal"].Width = (int)(([views get:@"panel1"].Width));
[views get:@"btnmenuprincipal"].Height = (int)((7 / 100.0 * height));
[views get:@"lblregistrarme"].Left = (int)(0.0);
[views get:@"lblregistrarme"].Top = (int)(([views get:@"btnmenuprincipal"].Top + [views get:@"btnmenuprincipal"].Height)+(1 / 100.0 * height));
[views get:@"lblregistrarme"].Width = (int)((100 / 100.0 * width));
[views get:@"lblregistrarme"].Height = (int)((5 / 100.0 * height));
[views get:@"label2"].Height = (int)((5 / 100.0 * height));
[views get:@"label2"].Width = (int)((100 / 100.0 * width));
[views get:@"label2"].Left = (int)(0.0);
[views get:@"label2"].Top = (int)((96 / 100.0 * height) - ([views get:@"label2"].Height));

}
@end